common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -809,3 +805,101 @@ def test_perform_problem_reset_and_navigate(self):
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1').add_children(
+                    create_multiple_choice_problem('Test Problem 1')
+                )
+            )
+        ).install()
+
+        self.progress_page = ProgressPage(self.browser, self.course_id)
+        self._setup_subsection()
+
+        # Auto-auth register for the course.
+        _auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+
+    def _setup_subsection(self):
+        """
+        Helper to set up a problem subsection as staff, then take
+        it as a student.
+        """
+        LogoutPage(self.browser).visit()
+        _auto_auth(self.browser, "STAFF_TESTER", "staff101@example.com", True, self.course_id)
+        self.course_outline.visit()
+        self.course_outline.open_subsection_settings_dialog()
+
+        self.course_outline.select_advanced_tab('hide_after_due_date')
+        self.course_outline.make_subsection_hidden_after_due_date()
+
+        LogoutPage(self.browser).visit()
+        _auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+        self.courseware_page.visit()
+
+        LogoutPage(self.browser).visit()
+
+    def test_subsecton_hidden_after_due_date(self):
+        """
+        Given that I am a staff member on the exam settings section
+        select advanced settings tab
+        When I Make the exam timed.
+        And I login as a verified student.
+        And visit the courseware as a verified student.
+        And I start the timed exam
+        Then I am taken to the exam with a timer bar showing
+        When I finish the exam
+        Then I see the exam submitted dialog in place of the exam
+        When I log back into studio as a staff member
+        And change the problem's due date to be in the past
+        And log back in as the original verified student
+        Then I see the exam or message in accordance with the hide_after_due setting
+        """
+        LogoutPage(self.browser).visit()
+        _auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+        self.courseware_page.visit()
+        self.assertFalse(self.courseware_page.has_past_due_date_message())
+
+        self.progress_page.visit()
+        self.assertEqual(self.progress_page.scores('Test Section 1', 'Test Subsection 1'), [(0, 1)])
+
+        LogoutPage(self.browser).visit()
+        _auto_auth(self.browser, "STAFF_TESTER", "staff101@example.com", True, self.course_id)
+        self.course_outline.visit()
+        last_week = (datetime.today() - timedelta(days=7)).strftime("%m/%d/%Y")
+        self.course_outline.change_problem_due_date(last_week)
+
+        LogoutPage(self.browser).visit()
```

Comment:
Nit: Is there a reason not to create a `self.logout_page` and use that repeatedly?